11-getting-started-with-Jupyter.Rmd

```diff
@@ -0,0 +1,350 @@
+to exist until it is deliberately deleted with R code, or when the Jupyter 
+notebook R session (*i.e.*, kernel) is stopped or restarted. It can also be 
+referenced in another distinct code cell. Together, this means that you could then write 
+a code cell further above in the notebook that references `y` and execute 
+it without error in the current session. This could also be done successfully in 
+future sessions if, and only if, you run the cells in the same non-conventional 
+order. However, it is difficult to remember this non-conventional order, and it 
+is not the order that others would expect your code to be executed in. An 
+example of this scenario is shown in figures 
+\@ref(fig:out-of-order-1) - \@ref(fig:out-of-order-3).
+
+Another thing that the flexibility of a Jupyter notebooks allows you to do is create 
+objects through running a cell, which later gets deleted. In such a scenario, that 
+object only exists for that one particular R session and would not exist again 
+if the notebook session was restarted and the notebook run again. If that 
+object was referenced in another cell in that notebook, an error would likely
+occur when the Jupyter notebook was run again in a new session.
+
+```{r out-of-order-1, echo = FALSE, fig.cap = "Code that was written out of order, but not yet executed.", fig.retina = 2, out.width="100%"}
+image_read("img/out-of-order-1.png") %>% 
+  image_crop("3632x800")
+```
+
+```{r out-of-order-2, echo = FALSE, fig.cap = "Code that was written out of order, and was executed using the run button in a non-linear error without error. The order of execution can be traced by following the numbers to the left of the code cells; their order indicates the order in which the cells were executed.", fig.retina = 2, out.width="100%"}
+image_read("img/out-of-order-2.png") %>% 
+  image_crop("3632x800")
+```
+
+```{r out-of-order-3, echo = FALSE, fig.cap = 'Code that was written out of order, and was executed in a linear order using "Restart Kernel and Run All Cells..." This resulted in an error at the execution of the second code cell and it failed to run all code cells in the notebook.', fig.retina = 2, out.width="100%"}
+image_read("img/out-of-order-3.png") %>% 
+  image_crop("3632x800")
+```
+
+These events may not negatively affect the current R session when 
+the code was being written, but as you might now see, they will likely lead to 
+errors when that notebook is run in a future session. Regularly, and 
+intentionally executing the entire notebook in a fresh R session will help guard 
+against this by letting a user know that some aspect of the code does not allow 
+the notebook to be run linearly. Knowing this sooner than later will allow the 
+user to fix the issue quickly and easily, ideally within that session. We 
+recommend doing this at least 2-3 times within any work session on a Jupyter 
+notebook.
+
+Note that a key thing about our best practice recommendation here is not just
```

Comment:
rather than "Note that a key thing about our best practice recommendation here" suggestion for rephrase: 
"We recommend as best practice to..."

Reply:
I kept "Note that a key thing" because I wanted to repeat this and draw attention to it. With the other sentence moved down I think it reads better as an emphasis now.